index.js

```diff
@@ -71,40 +75,44 @@ function trace(res, span) {
 }
 
 app.use((req, res, next) => {
-  if (req.method === "POST") {
-    const tracer = new Tracer({ service: "metaphysics" })
-    const span = tracer.startSpan("metaphysics.query")
-    const query = req.body.query.replace(/(\()([^\)]*)(\))/g, parse_args)
-    span.addTags({
-      resource: query,
-      type: "web",
-      "span.kind": "server",
-      "http.method": req.method,
-      "http.url": req.url,
-    })
-
-    assign(req, { span })
+  const tracer = new Tracer({ service: "metaphysics_testing" })
+  const span = tracer.startSpan("metaphysics_testing.query")
+  span.addTags({
+    type: "web",
+    "span.kind": "server",
+    "http.method": req.method,
+    "http.url": req.url,
+  })
 
-    res.on("finish", () => trace(res, span))
-    res.on("close", () => trace(res, span))
+  if (req.body && req.body.query) {
+    const query = drop_params(req.body.query)
+    span.addTags({ resource: query })
+  } else {
+    span.addTags({ resource: req.path })
   }
+
+  assign(req, { span })
+
+  res.on("finish", () => trace(res, span))
+  res.on("close", () => trace(res, span))
+
   next()
 })
 
 function wrapResolve(typeName, fieldName, resolver) {
-  return function (root, options, request) {
-    const parentSpan = request.span
-    const span = parentSpan.tracer().startSpan("metaphysics.resolver." + typeName + "." + fieldName,
+  return function (root, opts, req, { rootValue }) {
+    const parentSpan = rootValue.span
+    const span = parentSpan.tracer().startSpan("metaphysics_testing.resolver." + typeName + "." + fieldName,
       { childOf: parentSpan.context() })
     span.addTags({
       resource: typeName + ": " + fieldName,
       type: "web",
       "span.kind": "server",
     })
 
-    assign(request, { span })
+    rootValue.span = span // eslint-disable-line no-param-reassign
     const result = resolver.apply(this, arguments);
-    assign(request, { span: parentSpan })
+    rootValue.span = parentSpan // eslint-disable-line no-param-reassign
```

Comment:
I think @acjay asked this previously but it hasn’t been covered yet, what is this dance for where you first set `rootValue.span = span` and then revert back to the original `rootValue.span = parentSpan`? Maybe some code comments here could help?

Reply:
This wraps all resolvers so if there's a call to a subtype's resolver within a resolver we want it's parent to be the outer resolver.

Reply:
Right, I see. I do think that adding a comment that mentions something like “temporarily change context for nested resolvers” could help future readers of the code.